infer/SigSuggestion.cc

```diff
@@ -458,7 +459,8 @@ bool SigSuggestion::maybeSuggestSig(core::Context ctx, core::ErrorBuilder &e, un
                 chosenType = oldType;
             }
             if (!ctx.state.suggestRuntimeProfiledType || !chosenType->isUntyped()) {
-                fmt::format_to(ss, "{}: {}", argSym.argumentName(ctx), chosenType->show(ctx));
+            // TODO(jez) String substr is bad
+                fmt::format_to(ss, "{}: {}", argSym.argumentName(ctx).substr(1), chosenType->show(ctx));
```

Comment:
Dmitry: I was surprised to see that `argumentName` includes the `&`. It looks like our Loc for the block argument points to the whole thing: `&blk`. Can you think of a case where the implementation would be relying on this behavior? I'm going to try changing `argumentName` to drop the `&` for block arguments, which i think a better place to be regardless.

Reply:
Great question. I agree with you: it should include `&`. We should prepend `&` based on ArgInfo if we need it.

Reply:
(important edit, I forgot "not")